Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1500,8 +1537,41 @@ TEST_F(SerializerTest, DecodesListenResponseWithTargetChange) {
   proto.mutable_target_change()->add_target_ids(2);
   proto.mutable_target_change()->set_resume_token("resume_token");
 
-  SCOPED_TRACE("DecodesListenResponseWithTargetChange");
-  ExpectWatchChangeDeserializationRoundTrip(model, proto);
+  SCOPED_TRACE("DecodesListenResponseWithAddedTargetChange");
+  ExpectDeserializationRoundTrip(model, proto);
+}
+
+TEST_F(SerializerTest, DecodesListenResponseWithRemovedTargetChange) {
+  WatchTargetChange model(WatchTargetChangeState::Removed, {1, 2},
```

Comment:
The Objective-C version also checks the error case by setting a failed `Status`, can this piece be ported, too?

Reply:
Done.